src/main/scala/beam/agentsim/agents/vehicles/BeamVehicleAgent.scala

```diff
@@ -1,16 +1,28 @@
 package beam.agentsim.agents.vehicles
 
-import akka.actor.ActorRef
+import akka.actor.{ActorRef, Props}
 import beam.agentsim.events.resources.vehicle.GetVehicleLocationEvent
 import org.matsim.api.core.v01.Id
 import org.matsim.vehicles.Vehicle
 import akka.pattern.pipe
+import beam.agentsim.agents.InitializeTrigger
+
 import scala.concurrent.ExecutionContext.Implicits.global
 
 /**
   *
   * @author dserdiuk
   */
+object BeamVehicleAgent {
+
+  def props(vehicleId: Id[Vehicle], matSimVehicle: Vehicle, trajectory: Trajectory, powertrain: Powertrain) = {
```

Comment:
Yes, trajectory will be set by the driver during execution, not on construction.